doc/dev/osd_internals/snaps.rst

```diff
@@ -126,3 +126,84 @@ up to 8 prefixes need to be checked to determine all hobjects in a particular
+listnsnap output is as follows:
++------------+------------+-----------+-------------------------+
+| cloneid    | snaps      | size      | overlap                 |
++============+============+===========+=========================+
+| head       | -          | 4         |                         |
++------------+------------+-----------+-------------------------|
+
+After taking a snapshot (ID 1) and re-writing the first 2 bytes of the object,
+the clone created will overlap with the new head object in its last 2 bytes.
+
+object     content is: [BBAA]
+clone ID 1 content is: [AAAA]
+
++------------+------------+-----------+-------------------------+
+| cloneid    | snaps      | size      | overlap                 |
++============+============+===========+=========================+
+| 1          | 1          | 4         | [2~2]                   |
++------------+------------+-----------+-------------------------|
+| head       | -          | 4         |                         |
++------------+------------+-----------+-------------------------|
+
+Following the same process, by taking another snapshot (ID: 2)
+and this time re-writing only the first 1 byte of the object,
+the clone created (ID 2) will overlap with the new head object in its last 3 bytes.
+While the oldest clone (ID 1) will overlap with the newest clone in its last 2 bytes.
+
+object     content is: [CBAA]
+clone ID 2 content is: [BBAA]
+clone ID 1 content is: [AAAA]
+
++------------+------------+-----------+-------------------------+
+| cloneid    | snaps      | size      | overlap                 |
++============+============+===========+=========================+
+| 1          | 1          | 4         | [2~2]                   |
++------------+------------+-----------+-------------------------|
+| 2          | 2          | 4         | [1~3]                   |
++------------+------------+-----------+-------------------------|
+| head       | -          | 4         |                         |
++------------+------------+-----------+-------------------------|
+
+If the head object will be completely re-written by re-writing 4 bytes,
+the only existing overlap that will remain will be between the two clones.
+
+object     content is: [DDDD]
+clone ID 2 content is: [BBAA]
+clone ID 1 content is: [AAAA]
+
++------------+------------+-----------+-------------------------+
+| cloneid    | snaps      | size      | overlap                 |
++============+============+===========+=========================+
+| 1          | 1          | 4         | [2~2]                   |
++------------+------------+-----------+-------------------------|
+| 2          | 2          | 4         |                         |
++------------+------------+-----------+-------------------------|
+| head       | -          | 4         |                         |
++------------+------------+-----------+-------------------------|
+
+Lastly, when removing the last snap (ID 2), no overlapping intervals will remain:
```

Comment:
I think it's worth noting that that it happens not at the time the snapshot is removed, but potentially much later as part of snap trimming:
```suggestion
Lastly, after the last snap (ID 2) is removed and snaptrim kicks in, no overlapping intervals will remain:
```